src/diary_ui.cpp

```diff
@@ -168,10 +168,13 @@ void diary::show_diary_ui( diary *c_diary )
     enum class window_mode : int {PAGE_WIN = 0, CHANGE_WIN, TEXT_WIN, NUM_WIN, FIRST_WIN = 0, LAST_WIN = NUM_WIN - 1};
     window_mode currwin = window_mode::PAGE_WIN;
 
-    std::map<window_mode, int> selected = { {window_mode::PAGE_WIN, 0}, {window_mode::CHANGE_WIN, 0}, {window_mode::TEXT_WIN, 0} };
+    std::map<window_mode, int> selected = { {window_mode::PAGE_WIN, std::max<size_t>( 0, c_diary->pages.size() - 1 )}, {window_mode::CHANGE_WIN, 0}, {window_mode::TEXT_WIN, 0} };
```

Comment:
diary view will start on the latest page instead of the first page

Reply:
Won't the current code cause it to underflow to the maximum unsigned `size_t` when the size is zero?